residue_genomic_position_script.py

```diff
@@ -0,0 +1,266 @@
+import argparse
+from email import header
+from json import JSONDecodeError
+import logging
+import psutil
+
+import pandas as pd
+import requests
+from pandarallel import pandarallel
+import pyspark.sql.functions as f
+from pyspark.sql import SparkSession
+from pyspark.sql.types import StructField, StructType, IntegerType, StringType
+
+
+# Global configuration for Spark and Pandarallel.
+spark = SparkSession.builder.master('local[*]').getOrCreate()
+
+pandarallel.initialize(
+    nb_workers=psutil.cpu_count(),
+    progress_bar=True,
+)
+
+def main():
+
+    # PLIP INPUT (contain wanted data)
+    plip_json_input = (
+        # "gene_mapped_structures.json"
+        spark.read.json(args.plip_input)
+
+        .select("pdbStructureId", "chains", f.explode("compoundIds").alias("pdbCompoundId"))
+
+        .select("pdbStructureId", "pdbCompoundId", f.explode("chains").alias("chains"))
+
+        )
+
+    plip_json_input_v2 = (plip_json_input
+
+                        .withColumn("chainId", plip_json_input["chains.chainId"])
+
+                        .withColumn("geneId", plip_json_input["chains.geneId"])
+
+                        .withColumn("uniprotId", plip_json_input["chains.uniprot"])
+
+                        .drop("chains")
+
+                        )
+
+    # PLIP OUTPUT
+    plip_csv_output = (
+
+        # "output.csv"
+        spark.read.csv(args.plip_output, header=True, sep=",")
+
+        .withColumnRenamed("pdb_structure_id", "pdbStructureId")
+
+        .withColumnRenamed("compound_id", "pdbCompoundId")
+
+        .withColumnRenamed("prot_chain_id", "chainId")
+
+        )
+
+    # JOIN to have gene id (used for filter mapping file on the gene id)
+    plip_output_target_id = (
+
+        plip_json_input_v2
+
+        .join(plip_csv_output, on=["pdbStructureId", "chainId", "pdbCompoundId"])
+
+        .withColumnRenamed("interaction_type", "intType")
+
+        .withColumnRenamed("prot_residue_number", "protResNb")
+
+        .withColumnRenamed("prot_residue_type", "protResType")
+
+    )
+
+    # Target df
+    target_df = (
+
+        spark.read
+
+        .parquet("../targets")
+
+        .select("id", "genomicLocation")
+
+        .withColumn("chromosome", f.col("genomicLocation.chromosome"))
+        
+        .withColumnRenamed("id", "geneId")
+
+        .drop("genomicLocation")
+    )
```

Comment:
The above expression can be consolidated by saying:

```suggestion
    target_df = (
        spark.read.parquet(args.target)
        .select(
            f.col("id").alias('geneId'), 
            f.col("genomicLocation.chromosome")
        )
        .persist()
    )
```